java/code/src/com/suse/manager/webui/controllers/FormulaController.java

```diff
@@ -249,25 +260,25 @@ public String saveFormula(Request request, Response response, User user) {
         try {
             switch (type) {
                 case SERVER:
-                    Optional<MinionServer> minion = MinionServerFactory.lookupById(id);
+                    MinionServer minion = MinionServerFactory.lookupById(id).get();
                     try {
-                        FormulaUtil.ensureUserHasPermissionsOnServer(user, minion.get());
+                        FormulaUtil.ensureUserHasPermissionsOnServer(user, minion);
                     }
                     catch (PermissionException e) {
                         return deniedResponse(response);
                     }
-                    FormulaFactory.saveServerFormulaData(formData, MinionServerFactory.getMinionId(id), formulaName);
-                    saltApi.refreshPillar(new MinionList(minion.get().getMinionId()));
+                    FormulaFactory.saveServerFormulaData(formData, minion, formulaName);
+                    saltApi.refreshPillar(new MinionList(minion.getMinionId()));
```

Comment:
Nitpick: this could be moved inside the above `try` block ( As you did already for listing groups formulas). I would then be one logical block not interrupted by `catch` block.